lib/ansible/modules/database/postgresql/postgresql_publication.py

```diff
@@ -0,0 +1,286 @@
+        return result
+
+    query = """
+    SELECT schemaname, tablename FROM pg_publication_tables WHERE pubname = %(publication)s
+    """
+    cursor.execute(query, {'publication': publication})
+    result['tables'] = cursor.fetchall()
+    return result
+
+
+def publication_exists(cursor, publication):
+    query = "SELECT pubname FROM pg_publication WHERE pubname = %(publication)s"
+    cursor.execute(query, {'publication': publication})
+    return cursor.rowcount == 1
+
+
+def publication_delete(cursor, publication):
+    if publication_exists(cursor, publication):
+        query = "DROP PUBLICATION %s" % pg_quote_identifier(publication, 'publication')
+        cursor.execute(query)
+        return True
+    else:
+        return False
+
+
+def publication_modify_tables(cursor, publication, tables):
+    if not publication_exists(cursor, publication):
+        raise AssertionError("Modifying inexistant publication.")
+
+    table_quoted_list = []
+    for table in tables:
+        table_quoted_list.append('%s' % pg_quote_identifier(table, 'table'))
+
+    cursor.execute("ALTER PUBLICATION %s SET TABLE %s" % (pg_quote_identifier(publication, 'publication'),
+                   ', '.join(table_quoted_list)))
+    return True
+
+
+def publication_create(cursor, publication, tables):
+    if not publication_exists(cursor, publication):
+        query_fragments = ['CREATE PUBLICATION %s FOR' % pg_quote_identifier(publication, 'publication')]
+        if tables:
+            table_quoted_list = []
+            for table in tables:
+                table_quoted_list.append('%s' % pg_quote_identifier(table, 'table'))
+                query_fragments.append('TABLE %s' % ', '.join(table_quoted_list))
+        else:
+            query_fragments.append('ALL TABLES')
+        query = ' '.join(query_fragments)
+        cursor.execute(query)
+        return True
+    else:
+        publication_info = get_publication_info(cursor, publication)
+        # If publication switch from targeted tables to all tables (or invert), recreate it
+        if publication_info['alltables'] is True and tables or publication_info['alltables'] is False and not tables:
+            publication_delete(cursor, publication)
+            return publication_create(cursor, publication, tables)
+
+        return publication_modify_tables(cursor, publication, tables)
+
+
+def publication_matches(cursor, publication, owner):
+    if not publication_exists(cursor, publication):
+        return False
+    else:
+        publication_info = get_publication_info(cursor, publication)
+        if owner and owner != publication_info['owner']:
+            return False
+        else:
+            return True
+
+# ===========================================
+# Module execution.
+#
+
+
+def main():
+    argument_spec = pgutils.postgres_common_argument_spec()
+    argument_spec.update(dict(
+        name=dict(required=True),
+        database=dict(default="postgres"),
```

Comment:
Not sure why `libpq` [default value](https://www.postgresql.org/docs/10/static/libpq-connect.html#LIBPQ-CONNECT-DBNAME) is overridden for `database` parameter ?